lib/rubygems/gemcutter_utilities/webauthn_listener.rb

```diff
@@ -85,10 +85,17 @@ def root_path?(uri)
     end
 
     def parse_otp_from_uri(uri)
-      require "cgi"
+      query = uri.query
+      return unless query && !query.empty?
 
-      return if uri.query.nil?
-      CGI.parse(uri.query).dig("code", 0)
+      query.split('&').each do |param|
+        key, value = param.split('=', 2)
+        if key && value && Gem::URI.decode_www_form_component(key) == "code"
```

Comment:
If `value` is non-nil, `key` should be too.

```suggestion
        if value && Gem::URI.decode_www_form_component(key) == "code"
```